### PR TITLE
basics/templates: fix code example + subsections

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: basics/templates.md [hunk 1]
```diff
@@ -17,15 +17,18 @@ the function using the `!` operator:
...
 If no template parameter is given for a templated function the compiler
 tries to deduce the type using the input parameters with which the function
 is called:
 
     int a = 5; int b = 10;
     add(a, b); // T is to deduced to `int`
-    float c = 5.0f;
-    add(a, c); // ERROR: conflict because compiler
-               // doesn't know what T should be
+    float c = 5.0;
```
Comment (wilzbach) [inline basics/templates.md]: While it's for newbies, specifying the type twice is a redundant work, so either `float c = 5` or `auto c = 5f` would be enough. I left the `.0`, but I think `.f` is overly cautious.

File: basics/templates.md [hunk 2]
```diff
@@ -17,15 +17,18 @@ the function using the `!` operator:
     add!float(5.0f, 10.0f);
     add!Animal(dog, cat); // won't compile; Animal doesn't implement +
 
+### Automatic template inference
```
Comment (ZombineDev) [inline basics/templates.md]: IFTI (Implicit Function Template Instantiation"

File: basics/templates.md [hunk 3]
```diff
@@ -17,15 +17,18 @@ the function using the `!` operator:
...
+
 If no template parameter is given for a templated function the compiler
```
Comment (ZombineDev) [inline basics/templates.md]: Function templates have two parameter sets - the first is for compile-time arguments and the second is for run-time arguments. (Non-templated functions can accept only run-time arguments). If one ore more compile-time arguments are left unspecified when the function is called, the compiler tries to deduce them from the list of run-time arguments as the types of those arguments.

File: basics/templates.md [hunk 4]
```diff
@@ -17,15 +17,20 @@ the function using the `!` operator:
     add!float(5.0f, 10.0f);
     add!Animal(dog, cat); // won't compile; Animal doesn't implement +
 
-If no template parameter is given for a templated function the compiler
-tries to deduce the type using the input parameters with which the function
-is called:
+### IFTI (Implicit Function Template Instantiation)
```
Comment (JackStouffer) [inline basics/templates.md]: A less official sounding title would probably be clearer. i.e. "Implicit Template Parameters"

**Pull request merged**